packages/vue-app/template/utils.js

```diff
@@ -264,7 +264,8 @@ export function getLocation (base, mode) {
   if (mode === 'hash') {
     return window.location.hash.replace(/^#\//, '')
   }
-  if (base && path.indexOf(base) === 0) {
+  // To get matched with sanitized router.base add trailing slash
+  if (base && (path.endsWith('/') ? path : path + '/').startsWith(base)) {
```

Comment:
This is an inline patch. Not sure but may introduce inconsistencies. WDYT @pimlie @manniL?